diff -urp old/board/ti/panda/panda_mux_data.h new/board/ti/panda/panda_mux_data.h
--- old/board/ti/panda/panda_mux_data.h	2020-05-16 10:04:18.000000000 +0200
+++ new/board/ti/panda/panda_mux_data.h	2020-05-21 01:09:16.000000000 +0200
@@ -66,6 +66,28 @@ const struct pad_conf_entry core_padconf
 {GPMC_WAIT1,  (PTD | OFF_EN | OFF_PD | OFF_OUT_PTD | M3)},	/* gpio_62 */
 {FREF_CLK2_OUT, (PTU | IEN | M3)},				/* gpio_182 */
 
+        {GPMC_A19, (PTU | IEN | OFF_EN | OFF_PD | OFF_IN | M1)},        /* kpd_row7 */
+        {GPMC_A22, (M3)},                                               /* gpio_46 */
+        {GPMC_A25, (PTD | M3)},                                         /* gpio_49 */
+        {GPMC_NCS3, (IEN | M3)},                                        /* gpio_53 */
+        {UART2_CTS, (PTU | IEN | M0)},                                  /* uart2_cts */
+        {UART2_RTS, (M0)},                                              /* uart2_rts */
+        {UART2_RX, (PTU | IEN | M0)},                                   /* uart2_rx */
+        {UART2_TX, (M0)},                                               /* uart2_tx */
+        {MCSPI1_CLK, (IEN | OFF_EN | OFF_PD | OFF_IN | M0)},            /* mcspi1_clk */
+        {MCSPI1_SOMI, (IEN | OFF_EN | OFF_PD | OFF_IN | M0)},           /* mcspi1_somi */
+        {MCSPI1_SIMO, (IEN | OFF_EN | OFF_PD | OFF_IN | M0)},           /* mcspi1_simo */
+        {MCSPI1_CS0, (PTD | IEN | OFF_EN | OFF_PD | OFF_IN | M0)},      /* mcspi1_cs0 */
+        {MCSPI1_CS1, (PTD | IEN | OFF_EN | OFF_PD | OFF_IN | M3)},      /* mcspi1_cs1 */
+        {MCSPI1_CS2, (PTU | OFF_EN | OFF_OUT_PTU | M3)},                /* gpio_139 */
+        {MCSPI1_CS3, (PTU | IEN | M3)},                                 /* gpio_140 */
+        {SDMMC5_CLK, (PTU | IEN | OFF_EN | OFF_OUT_PTD | M0)},          /* sdmmc5_clk */
+        {SDMMC5_CMD, (PTU | IEN | OFF_EN | OFF_PD | OFF_IN | M0)},      /* sdmmc5_cmd */
+        {SDMMC5_DAT0, (PTU | IEN | OFF_EN | OFF_PD | OFF_IN | M0)},     /* sdmmc5_dat0 */
+        {SDMMC5_DAT1, (PTU | IEN | OFF_EN | OFF_PD | OFF_IN | M0)},     /* sdmmc5_dat1 */
+        {SDMMC5_DAT2, (PTU | IEN | OFF_EN | OFF_PD | OFF_IN | M0)},     /* sdmmc5_dat2 */
+        {SDMMC5_DAT3, (PTU | IEN | OFF_EN | OFF_PD | OFF_IN | M0)},     /* sdmmc5_dat3 */
+
 };
 
 const struct pad_conf_entry wkup_padconf_array_essential[] = {
@@ -74,6 +96,26 @@ const struct pad_conf_entry wkup_padconf
 {PAD0_SR_SDA, (PTU | IEN | M0)}, /* sr_sda */
 {PAD1_SYS_32K, (IEN | M0)},	 /* sys_32k */
 {PAD0_FREF_CLK3_OUT, (M0)}, /* fref_clk3_out */
+        {PAD0_SIM_IO, (IEN | M0)},              /* sim_io */
+        {PAD1_SIM_CLK, (M0)},                   /* sim_clk */
+        {PAD0_SIM_RESET, (M0)},                 /* sim_reset */
+        {PAD1_SIM_CD, (PTU | IEN | M0)},        /* sim_cd */
+        {PAD0_SIM_PWRCTRL, (M0)},               /* sim_pwrctrl */
+        {PAD1_FREF_XTAL_IN, (M0)},              /* # */
+        {PAD0_FREF_SLICER_IN, (M0)},            /* fref_slicer_in */
+        {PAD1_FREF_CLK_IOREQ, (M0)},            /* fref_clk_ioreq */
+        {PAD0_FREF_CLK0_OUT, (M2)},             /* sys_drm_msecure */
+        {PAD1_FREF_CLK3_REQ, M7},               /* safe mode */
+        {PAD0_FREF_CLK3_OUT, (M0)},             /* fref_clk3_out */
+        {PAD0_FREF_CLK4_OUT, (PTU | M3)},       /* led status_2 */
+        {PAD0_SYS_NRESPWRON, (M0)},             /* sys_nrespwron */
+        {PAD1_SYS_NRESWARM, (M0)},              /* sys_nreswarm */
+        {PAD0_SYS_PWR_REQ, (PTU | M0)},         /* sys_pwr_req */
+        {PAD1_SYS_PWRON_RESET, (M3)},           /* gpio_wk29 */
+        {PAD0_SYS_BOOT6, (IEN | M3)},           /* gpio_wk9 */
+        {PAD1_SYS_BOOT7, (IEN | M3)},           /* gpio_wk10 */
+
+        {PAD1_FREF_CLK4_REQ, (PTU | M3)},       /* led status_1 */
 
 };
 
