diff -urp old/etc/encoding-profiles.conf new/etc/encoding-profiles.conf
--- old/etc/encoding-profiles.conf	2020-05-01 23:48:35.000000000 +0200
+++ new/etc/encoding-profiles.conf	2020-05-01 23:54:25.000000000 +0200
@@ -142,7 +142,7 @@ profile = enc-a-ac3
 of = dvd
 ofopts-append = packetsize=2048
 ofopts-append = muxrate=10080000
-vf-add = fps=25
+#vf-add = fps=25
 audio-samplerate = 48000
 ovcopts = g=15,b=6000000,maxrate=9000000,minrate=0,bufsize=1835008
 
@@ -153,7 +153,7 @@ profile = enc-a-ac3
 of = dvd
 ofopts-append = packetsize=2048
 ofopts-append = muxrate=10080000
-vf-add = fps="24000/1001"
+#vf-add = fps="24000/1001"
 audio-samplerate = 48000
 ovcopts = g=18,b=6000000,maxrate=9000000,minrate=0,bufsize=1835008
 
@@ -172,7 +172,7 @@ ovcopts-append = refs=5
 [enc-to-nok-6300]
 profile-desc = "3GP for Nokia 6300"
 profile = enc-f-3gp
-vf-add = fps=25
+#vf-add = fps=25
 vf-add = lavfi=graph="scale=176:144"
 audio-samplerate = 16000
 audio-channels = 1
