diff -urp old/wscript new/wscript
--- old/wscript	2020-05-02 11:39:01.000000000 +0200
+++ new/wscript	2020-05-02 12:02:53.000000000 +0200
@@ -1026,6 +1026,7 @@ def configure(ctx):
     ctx.find_program(cc,          var='CC')
     ctx.find_program(pkg_config,  var='PKG_CONFIG')
     ctx.find_program(ar,          var='AR')
+    ctx.find_program(cc,          var='AS')
     ctx.find_program('rst2html',  var='RST2HTML',  mandatory=False)
     ctx.find_program('rst2man',   var='RST2MAN',   mandatory=False)
     ctx.find_program('rst2pdf',   var='RST2PDF',   mandatory=False)
@@ -1035,6 +1036,7 @@ def configure(ctx):
     ctx.add_os_flags('LIBRARY_PATH')
 
     ctx.load('compiler_c')
+    ctx.load('gas')
     ctx.load('waf_customizations')
     ctx.load('dependencies')
     ctx.load('detections.compiler_swift')
