diff -urp old/arch/arm/mach-omap2/sata.c new/arch/arm/mach-omap2/sata.c
--- old/arch/arm/mach-omap2/sata.c	2020-06-13 13:16:44.000000000 +0200
+++ new/arch/arm/mach-omap2/sata.c	2020-06-30 19:18:17.000000000 +0200
@@ -425,10 +425,10 @@ static int sata_phy_init(struct device *
 	clk_enable(spdata->ref_clk);
 
 	omap_ocp2scp_init(dev, spdata->ocp2scp3);
-/*
+
 	sata_dpll_config(dev, spdata->pll);
 	sata_dpll_wait_lock(dev, spdata->pll);
-*/
+
 	omap_sataphyrx_init(dev, spdata->phyrx);
 	sataphy_pwr_init();
 	sataphy_pwr_on();
