diff -ur old/Makefile new/Makefile
--- old/Makefile	2011-12-11 09:08:15.000000000 +0100
+++ new/Makefile	2011-12-11 09:26:09.000000000 +0100
@@ -516,6 +516,7 @@
 SRCS_MPLAYER-$(ESD)          += libao2/ao_esd.c
 SRCS_MPLAYER-$(FBDEV)        += libvo/vo_fbdev.c libvo/vo_fbdev2.c
 SRCS_MPLAYER-$(OMAPDRMEGL)   += libvo/vo_omap_drm_egl.c libvo/yuv420_to_nv12.S
+SRCS_MPLAYER-$(OMAPDCE)      += libmpcodecs/vd_omap_dce.c
 SRCS_MPLAYER-$(FFMPEG)       += libvo/vo_png.c
 SRCS_MPLAYER-$(GGI)          += libvo/vo_ggi.c
 SRCS_MPLAYER-$(GIF)          += libvo/vo_gif89a.c
diff -ur old/configure new/configure
--- old/configure	2011-12-11 09:08:15.000000000 +0100
+++ new/configure	2011-12-11 09:21:29.000000000 +0100
@@ -705,6 +705,7 @@
 _vesa=auto
 _fbdev=auto
 _omapdrmegl=no
+_omapdce=no
 _dvb=auto
 _dxr2=auto
 _dxr3=auto
@@ -1071,6 +1072,8 @@
   --disable-fbdev)      _fbdev=no       ;;
   --enable-omapdrmegl)  _omapdrmegl=yes ;;
   --disable-omapdrmegl) _omapdrmegl=no  ;;
+  --enable-omapdce)     _omapdce=yes   ;;
+  --disable-omapdce)    _omapdce=no    ;;
   --enable-dvb)         _dvb=yes        ;;
   --disable-dvb)        _dvb=no         ;;
   --enable-dxr2)        _dxr2=yes       ;;
@@ -4686,6 +4689,22 @@
 
 
 
+echocheck "omapdce"
+if test "$_omapdce" = auto ; then
+  _omapdce=no
+  linux && _omapdce=yes
+fi
+if test "$_omapdce" = yes ; then
+  def_omapdce='#define CONFIG_OMAP_DCE 1'
+  vomodules="omapdce $vomodules"
+else
+  def_omapdce='#undef CONFIG_OMAP_DCE'
+  novomodules="omapdce $novomodules"
+fi
+echores "$_omapdce"
+
+
+
 echocheck "DVB"
 if test "$_dvb" = auto ; then
   _dvb=no
@@ -7895,6 +7914,7 @@
 FASTMEMCPY = $_fastmemcpy
 FBDEV = $_fbdev
 OMAPDRMEGL = $_omapdrmegl
+OMAPDCE = $_omapdce
 FREETYPE = $_freetype
 FTP = $_ftp
 GIF = $_gif
@@ -8440,6 +8460,7 @@
 $def_dxr3
 $def_fbdev
 $def_omapdrmegl
+$def_omapdce
 $def_ggi
 $def_ggiwmh
 $def_gif
diff -ur old/libmpcodecs/vd.c new/libmpcodecs/vd.c
--- old/libmpcodecs/vd.c	2011-10-14 23:54:33.000000000 +0200
+++ new/libmpcodecs/vd.c	2011-12-11 09:25:12.000000000 +0100
@@ -58,6 +58,7 @@
 extern const vd_functions_t mpcodecs_vd_libdv;
 extern const vd_functions_t mpcodecs_vd_lzo;
 extern const vd_functions_t mpcodecs_vd_qtvideo;
+extern const vd_functions_t mpcodecs_vd_omap_dce;
 
 /* Please do not add any new decoders here. If you want to implement a new
  * decoder, add it to libavcodec, except for wrappers around external
@@ -110,6 +111,9 @@
 #ifdef CONFIG_QTX_CODECS
     &mpcodecs_vd_qtvideo,
 #endif
+#ifdef CONFIG_OMAP_DCE
+    &mpcodecs_vd_omap_dce,
+#endif
     /* Please do not add any new decoders here. If you want to implement a new
      * decoder, add it to libavcodec, except for wrappers around external
      * libraries and decoders requiring binary support. */
