diff -urp old/sgx/services4/system/omap4/sysutils_linux.c new/sgx/services4/system/omap4/sysutils_linux.c
--- old/sgx/services4/system/omap4/sysutils_linux.c	2020-04-22 14:02:52.000000000 +0200
+++ new/sgx/services4/system/omap4/sysutils_linux.c	2020-04-22 13:53:39.000000000 +0200
@@ -344,7 +344,7 @@ IMG_VOID DisableSGXClocks(SYS_DATA *psSy
 
 #if (defined(DEBUG) || defined(TIMING)) && !defined(PVR_NO_OMAP_TIMER)
 #if defined(PVR_OMAP_USE_DM_TIMER_API)
-#define	GPTIMER_TO_USE 11
+#define	GPTIMER_TO_USE 10
 /*!
 ******************************************************************************
 
